buildSrc/src/main/kotlin/Java9Modularity.kt

```diff
@@ -73,39 +75,42 @@ object Java9Modularity {
         }
     }
 
-    private fun Project.registerCompileModuleTask(taskName: String, compileTask: KotlinCompile, sourceFile: File, targetFile: File) =
+    private fun Project.registerCompileModuleTask(taskName: String, compileTask: KotlinCompile, sourceFile: File, targetDirectory: Provider<out Directory>) =
         tasks.register(taskName, JavaCompile::class) {
-            // Also add the module-info.java source file to the Kotlin compile task;
-            // the Kotlin compiler will parse and check module dependencies,
-            // but it currently won't compile to a module-info.class file.
-            compileTask.source(sourceFile)
-
-
             // Configure the module compile task.
-            dependsOn(compileTask)
             source(sourceFile)
-            outputs.file(targetFile)
             classpath = files()
-            destinationDirectory.set(compileTask.destinationDirectory)
-            sourceCompatibility = JavaVersion.VERSION_1_9.toString()
-            targetCompatibility = JavaVersion.VERSION_1_9.toString()
+            destinationDirectory.set(targetDirectory)
+            javaCompiler.set(
+                this@registerCompileModuleTask.the<JavaToolchainService>().compilerFor {
+                    languageVersion.set(JavaLanguageVersion.of(9))
```

Comment:
This makes it harder to build the project on Apple Silicon Mac machines, as the `foojay` plugin provides no JDK 9 toolchains for the Apple Silicon

Perhaps it's more convenient to use Java 11 here and Java 9 as the target.

Reply:
That's a valid point. This is fixed now.